apps/web/app/(app)/providers.tsx

```diff
@@ -3,13 +3,16 @@ import React from "react";
 import { SessionProvider } from "@/providers/SessionProvider";
 import { SWRProvider } from "@/providers/SWRProvider";
 import { StatLoaderProvider } from "@/providers/StatLoaderProvider";
+import { QueueProvider } from "@/providers/QueueProvider";
 
 export default function Providers(props: { children: React.ReactNode }) {
   return (
     <SWRProvider>
       {/* <GmailProvider> */}
       <SessionProvider>
-        <StatLoaderProvider>{props.children}</StatLoaderProvider>
+        <StatLoaderProvider>
+          <QueueProvider>{props.children}</QueueProvider>
```

Comment:
Hey @elie222 ,

Do you think it is Ideal to Wrap QueueProvied in Root?

Because we need archiveEmails, readEmails, labelEmails, deleteEmails functionality into a single page that is `/mail`. So I think we should use QueueProvied inside `/mail`  page to make our app less bulk. P.S just a suggestion

Reply:
Hey,

I missed this before. We actually use it in a few places. You can see emails in many different pages in the app so it makes sense to keep it at the `(app)/layout` level.